adapters/sharethrough/utils.go

```diff
@@ -0,0 +1,217 @@
+	jsonPayload, err := json.Marshal(strResp)
+	if err != nil {
+		return "", err
+	}
+
+	tmplBody := `
+		<img src="//b.sharethrough.com/butler?type=s2s-win&arid={{.Arid}}" />
+
+		<div data-str-native-key="{{.Pkey}}" data-stx-response-name="{{.StrRespId}}"></div>
+	 	<script>var {{.StrRespId}} = "{{.B64EncodedJson}}"</script>
+	`
+
+	if params.Iframe {
+		tmplBody = tmplBody + `
+			<script src="//native.sharethrough.com/assets/sfp.js"></script>
+		`
+	} else {
+		tmplBody = tmplBody + `
+			<script src="//native.sharethrough.com/assets/sfp-set-targeting.js"></script>
+	    	<script>
+		     (function() {
+		       if (!(window.STR && window.STR.Tag) && !(window.top.STR && window.top.STR.Tag)) {
+		         var sfp_js = document.createElement('script');
+		         sfp_js.src = "//native.sharethrough.com/assets/sfp.js";
+		         sfp_js.type = 'text/javascript';
+		         sfp_js.charset = 'utf-8';
+		         try {
+		             window.top.document.getElementsByTagName('body')[0].appendChild(sfp_js);
+		         } catch (e) {
+		           console.log(e);
+		         }
+		       }
+		     })()
+		   </script>
+		`
+	}
+
+	tmpl, err := template.New("sfpjs").Parse(tmplBody)
+	if err != nil {
+		return "", err
+	}
+
+	var buf []byte
+	templatedBuf := bytes.NewBuffer(buf)
+
+	b64EncodedJson := base64.StdEncoding.EncodeToString(jsonPayload)
+	err = tmpl.Execute(templatedBuf, struct {
+		Arid           template.JS
+		Pkey           string
+		StrRespId      template.JS
+		B64EncodedJson string
+	}{
+		template.JS(strResp.AdServerRequestID),
+		params.Pkey,
+		template.JS(strRespId),
+		b64EncodedJson,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return templatedBuf.String(), nil
+}
+
+func (u Util) getPlacementSize(formats []openrtb.Format) (height uint64, width uint64) {
+	biggest := struct {
+		Height uint64
+		Width  uint64
+	}{
+		Height: 1,
+		Width:  1,
+	}
+
+	for i := 0; i < len(formats); i++ {
+		format := formats[i]
+		if (format.H * format.W) > (biggest.Height * biggest.Width) {
+			biggest.Height = format.H
+			biggest.Width = format.W
+		}
+	}
+
+	return biggest.Height, biggest.Width
+}
+
+func (u Util) canAutoPlayVideo(userAgent string) bool {
+	return (u.isAndroid(userAgent) && u.isAtMinChromeVersion(userAgent)) ||
+		(u.isiOS(userAgent) &&
+			(u.isAtMinSafariVersion(userAgent) || u.isAtMinChromeIosVersion(userAgent))) ||
+		!(u.isAndroid(userAgent) || u.isiOS(userAgent))
+}
+
+func (u Util) isAndroid(userAgent string) bool {
+	isAndroid, err := regexp.MatchString("(?i)Android", userAgent)
+	if err != nil {
+		return false
+	}
+	return isAndroid
+}
+
+func (u Util) isiOS(userAgent string) bool {
+	isiOS, err := regexp.MatchString("(?i)iPhone|iPad|iPod", userAgent)
+	if err != nil {
+		return false
+	}
+	return isiOS
+}
+
+func (u Util) isAtMinChromeVersion(userAgent string) bool {
+	var chromeVersion int64
+	var err error
+
+	chromeVersionRegex := regexp.MustCompile(`Chrome\/(?P<ChromeVersion>\d+)`)
```

Comment:
Is it possible to move the MustCompile() calls here and in the subsequent functions to NewSharethroughBidder() to set these variables at startup so that they don't need to be done on every request?  This would probably involve adding fields in the _SharethroughAdapter_ struct to hold the values.